lore/developers/contributing/designs/network-design-v2.md

```diff
@@ -0,0 +1,34 @@
+---
+name: network design v2
+layout: default
+---
+
+# BonnyCI Cloud Network Design Changes
+
+## Problem Statement
+
+The original network topology was designed to be a 2 tier setup where the control-plane network hosted all service hosts and the head bastion, and the nodepool network hosted all nodepool slaves.  The two networks were able to communicate via a router that had interfaces on both.  This works well to isolate the nodepool slaves from the main production systems, but is relatively flat with regards to the bastion and prevents us from bringing up other isolated Zuul environments that are managed by a single bastion.  We need to rethink this a bit so that the bastion can be decoupled from the control-plane, as we want to have multiple environments running at once, each with their own control-plane.  Additionally, we need to think about conserving resources and moving things that can be shared among different environments to a network that can be reached by all, while preserving network isolation between them.  This includes things like the logs server and the IRC Bot.
```

Comment:
There's been a push to make all Markdown files have line breaks at 80 characters. Please update accordingly.